onnxruntime/python/tools/transformers/bert_perf_test.py

```diff
@@ -119,6 +121,57 @@ def create_session(model_path, use_gpu, provider, intra_op_num_threads, graph_op
 
     return session
 
+def numpy_type(torch_type):
+    type_map = {torch.float32: np.float32,
+                torch.float16: np.float16,
+                torch.int32: np.int32,
+                torch.int64: np.longlong}
+    return type_map[torch_type]
+
+def create_input_output_tensors(inputs, outputs, device):
+    input_tensors = {name: torch.from_numpy(array).to(device)
+                        for name, array in inputs.items()}
+    output_tensors = {name: torch.from_numpy(array).to(device)
+                        for name, array in outputs.items()}
+    return input_tensors, output_tensors
+
+def create_io_binding(sess, input_tensors, output_tensors):
+    io_binding = sess.io_binding()
+    for name, tensor in input_tensors.items():
+        io_binding.bind_input(name, tensor.device.type, 0,
+                                numpy_type(tensor.dtype), tensor.shape,
+                                tensor.data_ptr())
+    for name, tensor in output_tensors.items():
+        io_binding.bind_output(name, tensor.device.type, 0,
+                                numpy_type(tensor.dtype), tensor.shape,
+                                tensor.data_ptr())
+    return io_binding
+
+def onnxruntime_inference_with_io_binding(session, all_inputs, output_names, test_setting):
+
+    inputs = random.choice(all_inputs)
+    result = session.run(output_names, inputs)
+    outputs = {}
+    for i in range(len(output_names)):
+        outputs[output_names[i]] = result[i]
+
+    device = 'cuda' if test_setting.use_gpu else 'cpu'
+    input_tensors, output_tensors = create_input_output_tensors(inputs, outputs, device)
+    io_binding = create_io_binding(session, input_tensors, output_tensors)
+
+    # warm up
+    for iter in range(1):
+        session.run_with_iobinding(io_binding)
+
+    results = []
+    latency_list = []
+    max_iters = 10
```

Comment:
Should this be set to the length of all_inputs?

Reply:
this is just a repeat tests for the same input from line #152, so 10 times should be good enough.

Reply:
Currently this number of test cases could be decided by a user by using flag --samples (it defaults to 10 indeed), different numbers of samples will affect the statistics computation in run_one_test. Larger number sample will give more accurate statistics, but also more costly to run. 

Although IO binding will just bind to a random test case, I think we should still measure/gather a user-specified number of samples to generate the statistics.

Reply:
Actually, maybe the correct way to do this is to bind for each test case (put io binding in the measurement for loop), but just measure the time session.run_with_iobinding(io_binding)

Reply:
let me wrap it in a loop